### ISSUE TITLE
Polish renderer API

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2619]
We now have each renderer in its own package but the API could be improved.

The symptom of the issue manifests itself from `nyxt/renderer/foo` being required to use unexported symbols from the `nyxt` package. Find a below a list of those:

```
nyxt::*context-menu-commands*
nyxt::*headless-p*
nyxt::*interactive-p*
nyxt::*run-from-repl-p*
nyxt::*schemes*
nyxt::+renderer+
nyxt::active-buffer
nyxt::active-prompt-buffers
nyxt::buffer-hide
nyxt::fullscreen-p
nyxt::get-keymap
nyxt::javascript-error-handler
nyxt::list-commands
nyxt::make-channel
nyxt::maximized-p
nyxt::open-urls
nyxt::panel-buffers
nyxt::print-message
nyxt::prompt-buffer-p
nyxt::run-async
nyxt::status
nyxt::tls-help
nyxt::update-prompt-input
nyxt::webkit-web-context
nyxt::without-package-locks
```

As the support for more renderers grow, the API should support a mechanism for users to switch between them.

The following operation should be possible:

- Start Nyxt with a given renderer enabled.
- Quit Nyxt, without killing the inferior Lisp process.
- Switch renderer using a single function call.
- Start Nyxt with another renderer enabled.

It would be great if it'd be possible to set the renderer when spawning windows :)

Follow the relevant discussions in 937262e and #2612.

**Comment by Ambrevar** [comment]
Renderer switching is now possible from the same REPL instance. see c52e797abef80e646daaea72733cf44ac2fca43f.

**Comment by aadcg** [comment]
@Ambrevar can we get an example on how to switch renderer from the REPL somewhere (comment, docstring, etc).

Also, could you add a simple test that captures this invariant? Thanks.

**Comment by Ambrevar** [comment]
I've added an example in commit f5181edf520674cd79d1c3aa83d24bf54488afb3.
However it does not work anymore.  I tried commit c52e797abef80e646daaea72733cf44ac2fca43f of the pull request where it was tested, and it's failing there too.

Either I'm using the wrong invocation, or something else broke.  Possibly a Guix update broke it.

**Comment by aadcg** [comment]
@Ambrevar, hmm, that's interesting... Maybe WebKitGTK internals changed slightly.

Conceptually, I understand that live renderer switching is a goal for Nyxt. Nonetheless, on practical terms, as of today, I don't think this is particularly important. I think that such a feature would be of great use once we will support both CEF and WebKitGTK. 

Recall #2721, a bug introduced when live renderer switching was introduced. With that in mind, I'd suggest temporarily disabling this feature.

**Comment by aadcg** [comment]
> But #2721 is just as (un)important since it only occurs at the REPL on restart.

Correct, it's absolutely unimportant for end users! But an annoyance for developers :p

I agree that we can remove the tag for #2721. Still, I find it a bit odd that we have that issue deriving from the implementation of a feature that doesn't seem to be so important and it is currently broken for obscure reasons.

**Comment by Ambrevar** [comment]
To clarify, do you mean there is anything actionable we should do before 3.0 in that regard?

**Comment by aadcg** [comment]
Nothing actionable for 3.0!

In my view we should signal things as they come up, otherwise we may forget :p

In that sense, we could add a note in the codebase mentioning that this feature currently broken.
